components/gitpod-db/src/typeorm/migration/1652365883273-CloneUrlIndexed.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the Gitpod Enterprise Source Code License,
+ * See License.enterprise.txt in the project root folder.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { columnExists, indexExists } from "./helper/helper";
+
+export class CloneUrlIndexed1652365883273 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        const TABLE_NAME = "d_b_workspace";
+        const COLUMN_NAME = "cloneURL";
+        const TYPE_INDEX_NAME = "d_b_workspace_cloneURL_idx";
+
+        if (!(await columnExists(queryRunner, TABLE_NAME, COLUMN_NAME))) {
+            await queryRunner.query(
+                `ALTER TABLE
+                    ${TABLE_NAME}
+                ADD COLUMN
+                    ${COLUMN_NAME} VARCHAR(256)
+                GENERATED ALWAYS AS (
+                    context ->> "$.repository.cloneUrl"
```

Comment:
Sorry @svenefftinge :cry:  this does not work: we learned the hard way that indexes on generated columns do not work :disappointed:

Reply:
They basically appear to work, but queries on them return inconsistent results.

Reply:
oh, interesting. Where did we learn that?

Reply:
`workspaceinstance.phase`. IMO we still have that column, but don't use it for that reason. We have phasePersisted instead.
To be clear: the field itself works. But building an index based on that does return the weird results. Maybe we overlooked options back then, so might be worth investigating a bit.

Reply:
> indexes on generated columns do not work 

I couldn't find any sign on the internet that this doesn't work, but multiple articles recommending the approach [[1](https://dev.mysql.com/doc/refman/8.0/en/create-table-secondary-indexes.html)], [[2](https://www.compose.com/articles/mysql-for-json-generated-columns-and-indexing/)], [[3](https://medium.com/1mgofficial/mysql-json-support-virtual-columns-and-indexing-json-31df4cc1aa31)], [[4](https://stackoverflow.com/questions/68118107/how-to-create-a-virtual-column-to-index-json-column-in-mysql)].

Reply:
@svenefftinge We can test by writing queries against `phase`. i can do that later and share a loom :slightly_smiling_face:

Reply:
I tried the following which didn't reveal any discrepancies:
```
SELECT status->>'$.phase' as phase1, phase, count(id) from d_b_workspace_instance where creationTime>'2022-05-13' group by phase1, phase
```

Reply:
We decided to not use the GENERATED COLUMN of mysql, but rather use the `phasePersisted` way, i.e. we will have a  a redundant `cloneUrl` column which is updated on save in typeorm. This is a rather boring solution, but we can move forward with our goals.